src/h5web/providers/hsds/api.ts

```diff
@@ -36,14 +35,15 @@ import {
   convertHsdsType,
   parseComplex,
   convertHsdsAttributes,
+  assertHsdsEntity,
 } from './utils';
 import { buildEntityPath, getChildEntity } from '../../utils';
 
 export class HsdsApi implements ProviderAPI {
   public readonly filepath: string;
   private readonly client: AxiosInstance;
 
-  private readonly entities = new Map<string, Entity>();
+  private readonly entities = new Map<string, HsdsEntity>();
```

Comment:
The entities we store have IDs, so I replace types `HsdsDataset` and `HsdsGroup` with a more generic `HsdsEntity<Entity>`.